webapp/javascript/components/FlameGraph/FlameGraphRenderer.jsx

```diff
@@ -119,6 +62,20 @@ class FlameGraphRenderer extends React.Component {
     this.abortCurrentJSONController();
   }
 
+  getFlameBearerFromProps(props) {
+    const actualProps = props ?? this.props;
+    switch (actualProps.viewType) {
+      case 'single':
+        return actualProps.single.flamebearer;
+      case 'double':
+        return actualProps.comparison[actualProps.viewSide].flamebearer;
+      case 'diff':
+        return actualProps.diff.flamebearer;
+      default:
+        return null;
```

Comment:
I think by making the `FlameGraphRenderer` dumb this function will not be needed anymore since it would receive the flamebearer as `props.flamebearer` from it's parent but I'll keep this in mind for future changes and adjustments.